### ISSUE TITLE
[ENH] proliferation of Rockets

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#3854]
It would appear that the contants of the folder `transformations.panel.dictionary_based._rocket` are highly duplicative, with now six variants of the rocket algorithm in it?

```python
__all__ = [
    "Rocket",
    "MiniRocket",
    "MiniRocketMultivariate",
    "MiniRocketMultivariateVariable",
    "MultiRocket",
    "MultiRocketMultivariate",
]
```

Not saying this is in itself a problem, because we made the decision to allow contribution of algorithms as long as they are well described.

What I wonder, there seems to be a lot of copy-paste repetition, so:

* is there an opportunity to refactor and reduce multiplication, e.g., a common `numba` utility module
* is the citation trail accurate? With all the copy-pasting, which algorithm should be citing which?

**Comment by fkiraly** [comment]
FYI authors @angus924, @michaelfeil

Related question: could `MiniRocketMultivariateVariable` not simply be a pipeline of an unequal length transformer and `MiniRocketMultivariate`, or is there some `numba` magic going on?

**Comment by MatthewMiddlehurst** [comment]
Relevant discussion in #3786

**Comment by fkiraly** [comment]
sounds mysterious - perhaps worthwhile to add what @angus924 has said in some of the docstrings, e.g., which rocket to use. E.g., which have currently unresolved problems, etc. Perhaps even an issue to track the unresolved issues, we could leverage the wisdom of the crowd if they are just precisely formulated and in an open isse.

I have also seen users on various social channels being confused about there being too many rockets, so making the docstrings clearer in that respect would also help people to use the one that @angus924 currently considers the "right" one.

**Comment by michaelfeil** [comment]
I agree, I added the interface. This could be easier for users, e.g. only a`MiniRocket` class. This would require according check functions for univariate / multivariate / unequal length checks in the `fit(X)` and `transform(X)`. 

I added the extra `MiniRocketMultivariateVariable` class, as it uses the sktime inner type of `df-list`, has different numba functions & signature.

**Comment by fkiraly** [comment]
@michaelfeil, I was referring more to the backend, i.e., the copy-pasting of the `numba` code.

Agree that the number of classes could also be reduced, but it's primarily the multiplication of the internal code that I think we need to address - DRY is more severely violated there.

**Comment by TonyBagnall** [comment]
hi, I have talked to @angus924 (code owner) about the multiple rocket transforms, and the original reason was to do with a problem he had with the multivariate version being much less efficient than the univariate one. It is something we will look at with him when time allows.
